### DISCUSSION TITLE
How do I set up OrchardCore for horizontal scaling in kubernetes?

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#15664] [Q&A] [answered]
For horizontal scaling to work, we require the containers to be stateless, however, whenever we recreate the container, we can't run auto-setup again as the tables are already created and hence the App_Data folder isn't created when a service restarts.

How do I go about handling this?

**Comment by Skrypt** [comment, marked as answer]
We have services that makes use of the Azure Blob Storage for this. So, basically, do not use the local App_Data folder.

**Reply by weirdyang** [reply]
I tried using the database shell configuration but unable to get it to work with auto setup, https://github.com/OrchardCMS/OrchardCore/issues/15665

**Comment by Piedone** [comment]
Alternatively, you can mount an Azure Files drive.

**Reply by weirdyang** [reply]
Do you mind elaborating on this? thanks.

**Reply by Piedone** [reply]
So, you can mount a shared Azure Files drive to your containers, serving as a durable storage. You [can then configure OC](https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/#orchard_app_data-environment-variable) to store the App_Data folder on that drive, by using a different path than the default one. This way, you can keep e.g. tenant settings even between deployments.

However, for storing shell configurations, the DB is better. This mounted drive approach can work well for Media though, and especially for the webroot folder (see https://github.com/OrchardCMS/OrchardCore/issues/15016#issuecomment-1917924446).

**Reply by weirdyang** [reply]
thanks!
